openapi-cli/src/main/java/io/ballerina/openapi/generators/client/FunctionSignatureGenerator.java

```diff
@@ -407,8 +406,7 @@ private Node getHeaderParameter(Parameter parameter, NodeList<AnnotationNode> pa
                         createToken(EQUAL_TOKEN), literalValueToken);
             } else {
                 BuiltinSimpleNameReferenceNode typeName = createBuiltinSimpleNameReferenceNode(null,
-                        createIdentifierToken(convertOpenAPITypeToBallerina(
-                                parameter.getSchema().getType().trim()) + "?"));
+                        createIdentifierToken(type + "?"));
```

Comment:
Please use [this](https://github.com/ballerina-platform/ballerina-openapi/blob/0a16952212bb652feda74a85915c757f0ae96b08/openapi-cli/src/main/java/io/ballerina/openapi/generators/GeneratorConstants.java#L248) defined constant in GeneratorConstants for "?".